crd-generator/api/src/main/java/io/fabric8/crd/generator/CRDGenerator.java

```diff
@@ -68,6 +69,11 @@ public CRDGenerator withOutput(CRDOutput<? extends OutputStream> output) {
     return this;
   }
 
+  public CRDGenerator withParallelGenerationEnabled(boolean parallel) {
```

Comment:
Just an idea: what if the ExecutorService is passed instead? That would give a better control over the execution.

Reply:
In general I think this makes sense. 
In this specific case, given the crd generation is a short-running task, I don't believe it's worth to manage the thread pool at higher level